### ISSUE TITLE
The Matrix Awakens textures are missing/broken/don't load (Compiled from UE5.1)

### ISSUE THREAD

**Issue by p0ryae** [HansKristian-Work/vkd3d-proton#1229]
When launching Matrix Awakens by using proton, the game is pretty smooth and the character skins and NPC skins load, but when it comes to loading the world textures, car textures and road textures the game fails to do so.

![image](https://user-images.githubusercontent.com/59381835/189542840-b2a951ae-ea54-4f20-83da-a7fb1ef1fb41.png)

- Xserver version: 1.21.1.4
- DXVK version:  vkd3d-proton-2.6.0
- Wine/Proton version: Proton Experimental Bleeding-Edge-7.0-245

**More Information:** After a bit of research and inspection, it seems VKD3D-Proton does not support this. The shader model 6.6 code path seems to be broken in the demo. Personally think it depends on NvAPI to use 64bit atomics in shaders.

**Comment by K0bin** [comment]
> Personally think it depends on NvAPI to use 64bit atomics in shaders.

Exactly that.

**Comment by DashCruft** [comment]
But shouldn't https://github.com/jp7677/dxvk-nvapi take care of that?

**Comment by K0bin** [comment]
No, the vkd3d-proton shader compiler(s) would have to be aware of it.

**Comment by DashCruft** [comment]
At the current stage of the development, is it possible to see the possibility of it being a thing near the future?

**Comment by HansKristian-Work** [comment]
Not really. We support SM 6.6 as the standard way to use 64-bit atomics.

**Comment by nmercaldo** [comment]
Hey @DashCruft 

I've got a project (with patches ) here for getting the city sample up and running under wine + vkd3d-proton.

The only missing pieces are some shader overrides you will need to do. (A couple infinite for loops)

Basically in UE5 dev they have enabled sm 6.6 features which allow the engine to avoid using nvidia extensions

**Comment by DashCruft** [comment]
> Hey @DashCruft
> 
> I've got a project (with patches ) here for getting the city sample up and running under wine + vkd3d-proton.
> 
> https://github.com/ParaLock/UE5OnWine
> 
> The only missing pieces are some shader overrides you will need to do. (A couple infinite for loops)
> 
> Basically in UE5 dev they have enabled sm 6.6 features which allow the engine to avoid using nvidia extensions

Omg...That's amazing. Going to test it right now without a doubt.

**Comment by ParaLock** [comment]
@DashCruft,

Hey! Its nmercaldo again (this time from my non-school git account). If you are not using a newer build of the engine you will want to use the sm 6.6 patch from this commit:

https://github.com/ParaLock/UE5OnWine/commit/f8dc5f819b625e6eeaeedd4cc6f52c4409132dab

I removed those fixes from later versions of the file due to them being made in the dev branch.

**Comment by DashCruft** [comment]
Hm there's a bit of confusion I got. How many I use the patch with the current build of the game that I have?

**Comment by ParaLock** [comment]
Hey @DashCruft,

Are you compiling the engine from source? If so try applying the patches as described in the repo. Unfortunately if you are not compiling the entire engine from source your kinda out of luck, at least until UE removes vendor extension use on their end.

**Comment by DashCruft** [comment]
Thanks. I will re-compile it with the patch as you describe and keep this thread updated.

**Comment by ParaLock** [comment]
@DashCruft best of luck! Unfortunately I will be unable to provide additional support but I can definitely say that I was able to run city sample with nvidia hardware. :)

**Comment by DashCruft** [comment]
@ParaLock When applying the patch (or even other patches) to the cloned unreal engine repo, I come up with this error:

![image](https://user-images.githubusercontent.com/59381835/191895337-ca76d468-c82d-41d6-87f6-eded92f2a87d.png)

Something doesn't seem to be applying right.

**Comment by HansKristian-Work** [comment]
> The only missing pieces are some shader overrides you will need to do. (A couple infinite for loops)

Could you elaborate what this means? Is there a dxil-spirv bug you're trying to work around here?

**Comment by Blisto91** [comment]
@nmercaldo Friendly ping

**Comment by p0ryae** [comment]
I think you should ping his main account for faster response.

@ParaLock friendly ping

**Comment by HansKristian-Work** [comment]
Closing as non-actionable.
